src/environmentd/src/lib.rs

```diff
@@ -389,11 +389,61 @@ impl Listeners {
             );
             computed
         };
+        // Determine the maximum wait time when doing a 0dt deployment.
+        let with_0dt_deployment_max_wait = {
+            let default = config
+                .system_parameter_defaults
+                .get(WITH_0DT_DEPLOYMENT_MAX_WAIT.name())
+                .map(|x| {
+                    Duration::parse(VarInput::Flat(x)).map_err(|err| {
+                        anyhow!(
+                            "failed to parse default for {}: {:?}",
+                            WITH_0DT_DEPLOYMENT_MAX_WAIT.name(),
+                            err
+                        )
+                    })
+                })
+                .transpose()?;
+            let ld = get_ld_value(
+                WITH_0DT_DEPLOYMENT_MAX_WAIT.name(),
+                &remote_system_parameters,
+                |x| {
+                    Duration::parse(VarInput::Flat(x)).map_err(|err| {
+                        format!(
+                            "failed to parse LD value {} for {}: {:?}",
+                            x,
+                            WITH_0DT_DEPLOYMENT_MAX_WAIT.name(),
+                            err
+                        )
+                    })
+                },
+            )?;
+            let catalog = openable_adapter_storage
+                .get_0dt_deployment_max_wait()
+                .await?;
+            let computed = catalog
+                .or(ld)
+                .or(default)
+                .unwrap_or(WITH_0DT_DEPLOYMENT_MAX_WAIT.default().clone());
```

Comment:
I see that this matches `enable_0dt_deployment`, but now that I'm thinking about it, I'm surprised to see `catalog` having a higher preference than `ld`. If we update the LD flag, wouldn't we want to use that value before we have a chance to also update the `catalog` value?

Reply:
I think normally the LD flag also gets synced to the catalog, but this is to make sure that what is explicitly set in the catalog has precedence.

Now that _I_ think about it, though, I'm also not sure anymore. Maybe @benesch? :sweat_smile:

Reply:
Hmm, I think we may have got the order wrong for `enable_0dt_deployment` too. While `txn_wal_tables` uses a different naming scheme, this is the order of priority:

1. CLI flag.
2. Value gotten from Launch Darkly sync.
3. Value found in the catalog.
4. Server configuration parameter default value.

> I think normally the LD flag also gets synced to the catalog

This does happen, but it happens at some point after this line.

Reply:
> but this is to make sure that what is explicitly set in the catalog has precedence.

I think the LD value should have precedence to what is explicitly set in the catalog. If we want to change this value before startup, we can only change it in LD in which case it will not be explicitly set in the catalog until later in the startup process.

Reply:
Concur. I think I just got this wrong. Sorry!

Reply:
will fix this up and then also fix up the `enable_0dt_deployment` flag 👌

Reply:
pushed up a commit that fixes both, please take yet another look :see_no_evil: